pages/index.tsx

```diff
@@ -48,44 +51,50 @@ const Blurb = styled.div`
 `;
 
 export default function Index() {
-  const [tabIndex, setTabIndex] = useState(0);
   const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
-  const [selectedIndex, setSelectedIndex] = useState(0);
+  const [selectedMenuItem, setSelectedMenuItem] = useState<string>(
+    "General Info"
+  );
   const { address: collAddress } = Collateral.useContainer();
   const { address: tokenAddress } = Token.useContainer();
   const { contract: weth } = WethContract.useContainer();
+  const { empAddress } = EmpAddress.useContainer();
 
   const isYieldToken =
     tokenAddress &&
     Object.keys(YIELD_TOKENS).includes(tokenAddress.toLowerCase());
 
-  const options = ["General Info", "Manage Position", "All Positions"];
-
-  if (isYieldToken) {
-    options.push("yUSD Yield");
-  }
+  let options = ["General Info", "Manage Position", "All Positions"];
+  const buildOptionsList = () => {
+    // If it is weth collateral contract then add the weth option.
+    if (weth && collAddress?.toLowerCase() == weth.address.toLowerCase()) {
+      options.push("Wrap/Unwrap WETH");
+    }
+
+    // If it is a yield token then add the yUSD yield and analytics tabs.
+    if (isYieldToken) {
+      options = options.concat(["yUSD Yield", "Analytics"]);
+    }
+
+    // Update selected page if the user toggles between EMPs while selected on
+    // invalid pages (i.e on Wrap/Unwrap then moves to uUSDrBTC).
+    if (options.indexOf(selectedMenuItem) == -1) {
+      setSelectedMenuItem("General Info");
+    }
+  };
+  buildOptionsList();
```

Comment:
Why do you call this in the mainline render _and_ in the use effect? I haven't deeply absorbed the flow, but this just struck me as a little off.

Reply:
Ye, good point. I've refactored this. it's cleaner now.